File: client/src/app/model/room.ts
interface IRoom {
    id?: number;
    name?: string;
    number?: number;
    visitors?: number;
    price?: number;
    rating?: number;
    description?: string;
}
export class Room implements IRoom {
    id:number;
    name:string;
    number:number;
    visitors:number;
    price:number;
    rating:number;
    description:string;

    constructor(obj:IRoom = {} as IRoom) {
        let {
            name = '0',
            number = 0,
            visitors = 0,
            price = 0,
            rating = 0,
            description = '0'
            }:IRoom = obj as IRoom;
        this.name = name;
        this.number = number;
        this.visitors = visitors;
        this.price = price;
        this.rating = rating;
        this.description = description;
    }
}

File: src/main/java/by/kohanova/model/Room.java
package by.kohanova.model;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "rooms")
public class Room implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    @Column
    public String name;

    @Column
    public int visitors;

    @Column
    public double price;

    @Column
    public double rating;

    @Column
    public String description;

    @OneToMany(mappedBy = "room")
    private List<News> news;

    @Override
    public boolean equals(Object obj) {
        if (obj == null)
            return false;
        if (!(obj instanceof User))
            return false;
        User other = (User) obj;
        return id != null && id.equals(other.id);
    }

    @Override
    public int hashCode() {
        return id == null ? 0 : id.hashCode();
    }
}
